python/langsmith/utils.py

```diff
@@ -23,6 +28,63 @@ class LangSmithConnectionError(Exception):
     """Couldn't connect to the LangSmith API."""
 
 
+class DictMixin(dict):
```

Comment:
If we're making such an involved mixing might as well use dataclass ?

Reply:
Dataclasses have annoying inheritance patterns but I could try not using `RunBase` or anything like that for these